charts/external-dns/templates/deployment.yaml

```diff
@@ -16,9 +16,7 @@ spec:
       {{- include "external-dns.selectorLabels" . | nindent 6 }}
   strategy:
     {{- toYaml .Values.deploymentStrategy | nindent 4 }}
-  {{- with .Values.revisionHistoryLimit }}
-  revisionHistoryLimit: {{ . }}
-  {{- end }}
+  revisionHistoryLimit: {{ .Values.revisionHistoryLimit }}
```

Comment:
```suggestion
  {{- if or .Values.revisionHistoryLimit (eq .Values.revisionHistoryLimit 0) }}
  revisionHistoryLimit: {{ .Values.revisionHistoryLimit }}
  {{- end }}
```

Reply:
This works fine for the unset or the greater than 0 cases, but for 0 I get:
```
Error: template: external-dns/templates/deployment.yaml:19:42: executing "external-dns/templates/deployment.yaml" at <eq .Values.revisionHistoryLimit 0>: error calling eq: incompatible types for comparison

Use --debug flag to render out invalid YAML
```
If I use `| int` then 0 works, but then the unset case becomes 0 too and an empty value gets templated in. Like I said, this isn't straightforward :wink:

Reply:
Did you test it with `0` or `"0"`? The latter would be expected to error.

Reply:
`0`, no quotes. I'm testing by running `helm template`, changing the `values.yaml` and checking the rendered output. I'm using helm 3.13.2.